### PR TITLE
extract formatter from adapters

### PR DESCRIPTION
**Pull request by alanhala**
## Summary
- Extracted the instantiation of the formatter to the paginate module
- Now we send the adapter to the formatter. The formatter asks the adapter for the content, the total count and count.

We have to extract the repeated logic, like the instantiation, from the adaptors to a superclass maybe

### CODE REVIEW

File: lib/wor/paginate/adapters/active_model.rb [hunk 1]
```diff
@@ -5,12 +5,24 @@ module Wor
...
+
+        def initialize(content, page, limit)
+          @content = content
+          @page = page
+          @limit = limit
+        end
+
+        def adapt?
+          return false if @content.is_a? Enumerable
+          @content.respond_to? :count # no
```
Comment (hdf1986) [inline lib/wor/paginate/adapters/active_model.rb]: Attr_readr plz

File: lib/wor/paginate/adapters/iterable.rb [hunk 2]
```diff
@@ -4,15 +4,30 @@ module Wor
...
+        attr_reader :page
+
+        def initialize(content, page, limit)
+          @content = content
+          @page = page
+          @limit = limit
+        end
+
+        def adapt?
+          @content.respond_to? :to_a
```
Comment (hdf1986) [inline lib/wor/paginate/adapters/iterable.rb]: Attr_readr plz

File: lib/wor/paginate/adapters/iterable.rb [hunk 3]
```diff
@@ -4,15 +4,30 @@ module Wor
...
+        end
+
+        def paginated_content
+          return @paginated_content if @paginated_content
```
Comment (hdf1986) [inline lib/wor/paginate/adapters/iterable.rb]: Well, you know, i love attr_readers

**Comment by alanhala** [comment]
@hdf1986 I don't like the idea of making public every instance variable of a class. I prefer having those variables as private and accessing them inside a class with the @, and then if someone needs to have access to that variable, I make the getter. But if you all agree with changing that, I create the getter for each one

**Pull request merged**